api/src/main/java/javax/config/Config.java

```diff
@@ -77,10 +78,24 @@
  * @author <a href="mailto:rsmeral@apache.org">Ron Smeral</a>
  * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
  * @author <a href="mailto:gunnar@hibernate.org">Gunnar Morling</a>
+ * @author <a href="mailto:mail@sebastian-daschner.com">Sebastian Daschner</a>
  *
  */
 public interface Config {
 
+    /**
+     * Return the resolved String property value for the specified property name
+     * from the underlying {@link ConfigSource ConfigSources}.
+     *
+     * If this method gets used very often then consider to locally store the configured value.
+     *
+     * @param propertyName
+     *             The configuration propertyName.
+     * @return the resolved property value as String
+     * @throws java.util.NoSuchElementException if the property isn't present in the configuration.
+     */
+    String getValue(String propertyName);
```

Comment:
General question: should we provide `Optional<String> value(String propertyName)` in addition. This could even have a default implementation like

```
default Optional<String> value(final String propertyName) { 
    return Optional.ofNullable(getValue(propertyName));
}
```

I really like this pattern and a developer can choose what method he wants to use.

Reply:
mmh, the default implementation won't work since `getValue` throws a `java.util.NoSuchElementException` if the value is not present. The `Optional` method should return an empty `Optional` in that case.

What do you think?